pkg/secretlock/secretlock.go

```diff
@@ -0,0 +1,28 @@
+/*
+Copyright SecureKey Technologies Inc. All Rights Reserved.
+SPDX-License-Identifier: Apache-2.0
+*/
+
+package secretlock
+
+// EncryptRequest for encrypting remote kms requests
+type EncryptRequest struct {
+	Plaintext                   string
+	AdditionalAuthenticatedData string
+}
+
+// DecryptRequest for decrypting remote kms requests
+type DecryptRequest struct {
+	Ciphertext                  string
+	AdditionalAuthenticatedData string
+}
+
+// EncryptResponse for receiving encryption response from remote kms requests
+type EncryptResponse struct {
+	Ciphertext string
```

Comment:
Are we expecting more field in this ? If not, directly using the field would be more useful as we can avoid working with structs.

Reply:
I followed Tink's AEADEnvelope encrypt/decrypt requests used to talk with external secret locks. Technically these are cloud requests/responses. In the case of this PR and the next one, we will mimic cloud calls for our default (local) secret lock but this will be eventually used by GCP/AWS/Hashicorp as well, so it's safer to match field names.